testing/integration/endpoint_security_test.go

```diff
@@ -894,3 +901,536 @@ func TestForceInstallOverProtectedPolicy(t *testing.T) {
 	out, err := fixture.Exec(ctx, args)
 	require.Errorf(t, err, "No error detected, command output: %s", out)
 }
+
+func TestInstallDefendWithMTLSandEncCertKey(t *testing.T) {
+	stack := define.Require(t, define.Requirements{
+		Group: Fleet,
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		// Only supported on Linux
+		OS: []define.OS{{Type: define.Linux}},
+	})
+
+	ctx := context.Background()
+	testUUID := uuid.Must(uuid.NewV4()).String()
+	policyID := "mTLS-defend-" + testUUID
+
+	fleetServerURL, err := fleettools.DefaultURL(ctx, stack.KibanaClient)
+	require.NoError(t, err, "failed getting Fleet Server URL")
+
+	defaultFleetHost := fleetServerURL[8:]
+	fleethostWrong, err := url.Parse("https://fixme.elastic.co:443")
+	require.NoError(t, err, "failed parsing fleethostWrong")
+
+	// ================================= proxy =================================
+	mtlsCLI, mtlsPolicy, oneWayTLSPolicy, proxyCLI, proxyPolicymTLS, proxyPolicyOneWayTLS := prepareProxies(t, fleethostWrong, defaultFleetHost)
+
+	// =================== Prepare fleet hosts and proxies ===================
+
+	// create mTLS proxy
```

Comment:
Is it possible to create the policies and enrollment tokens as part of each testcase instead of preparing a bunch of different policies and tokens upfront and only have the testcases specify the token use for enrollment? I know that there's a 1-1 relationship between the enrollment token and the policy but this setup makes the test kinda hard to follow:
- the policy definition is far removed from the assertions on the agent behavior
- each test case enroll exactly once with a given token and the a big assertion function is called to wait for the policy to apply and check the results
- it's kinda difficult to follow which test uses which policy (I know, I know, enrollment token... but still...)

Reply:
I did that way to minimise the code for each test case and let it be reusable.
Just the `enrollmentTokenOneWayProxyResp` is used only once. the others are used at least by 2 tests.

I get your point, but I believe it'd be worse having each test to set it up. It'd be more repetitive as each test would need to set it up and the test cases would be a lot longer, making it harder to see what each test do.